app/src/main/java/com/example/androidlearningapp/movies/data/api/MovieRoomCacheManager.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.androidlearningapp.movies.data.api
+
+import android.content.Context
+import com.example.androidlearningapp.movies.data.AppRoomDatabase
+import com.example.androidlearningapp.movies.entity.Movie
+import com.example.androidlearningapp.movies.entity.MovieRoomEntity
+
+class MovieRoomCacheManager(context: Context) : MovieCacheApi {
+
+    private val appRoomDataBaseInstance = AppRoomDatabase.getInstance(context)
+
+    override fun putEntries(movies: List<Movie>) {
+        appRoomDataBaseInstance?.movieDao()?.insertAll(*parseMoviesToMovieRoomEntities(movies))
+    }
+
+    override fun getEntries(): MutableList<Movie> {
+        return parseMovieRoomEntitiesToMovies(appRoomDataBaseInstance?.movieDao()?.getAll())
+    }
+
+    override fun removeEntries() {
+        appRoomDataBaseInstance?.movieDao()?.removeAll()
+    }
+
+    private fun parseMoviesToMovieRoomEntities(movies: List<Movie>): Array<MovieRoomEntity> {
+        val movieRoomEntityList: MutableList<MovieRoomEntity> = mutableListOf()

```

Comment:
return movies.map { MovieRoomEntity(
                    name = it.name,
                    nameEng = it.nameEng,
                    premiere = it.premiere,
                    description = it.description,
                    cover = it.image,
                    id = null) }
.toTypedArray()

Reply:
done